src/lib/UI/Service/IconResolver.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\UI\Service;
+
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Symfony\Component\Asset\Packages;
+
+abstract class IconResolver
+{
+    protected const DEFAULT_IDENTIFIER = 'default-config';
+    protected const ICON_KEY = 'thumbnail';
+
+    private ConfigResolverInterface $configResolver;
+
+    protected Packages $packages;
+
+    public function __construct(ConfigResolverInterface $configResolver, Packages $packages)
+    {
+        $this->configResolver = $configResolver;
+        $this->packages = $packages;
+    }
+
+    protected function getIcon(string $format, string $identifier): string
+    {
+        $icon = $this->resolveIcon($format, $identifier);
+        $fragment = null;
+        if (strpos($icon, '#') !== false) {
+            [$icon, $fragment] = explode('#', $icon);
+        }
+
+        return $this->packages->getUrl($icon) . ($fragment ? '#' . $fragment : '');
+    }
+
+    private function resolveIcon(string $format, string $identifier): string
+    {
+        $parameterName = $this->getConfigParameterName($format, $identifier);
+        $defaultParameterName = $this->getConfigParameterName($format, static::DEFAULT_IDENTIFIER);
+
+        if ($this->configResolver->hasParameter($parameterName)) {
+            $config = $this->configResolver->getParameter($parameterName);
+        }
+
+        if (
+            (empty($config) || empty($config[static::ICON_KEY]))
```

Comment:
Why not:
```suggestion
            (empty($config) || empty($config[self::ICON_KEY]))
```
?

Reply:
This is because of how self and static work. If you use self the value always will be `thumbnail`. You will not be able to overwrite it in the parent class.